addon/chrome/content/preferences/appearance.js

```diff
@@ -59,6 +59,32 @@ var gAppearancePane = {
 
     gPrefWindow.removeChild("obs_smoothScroll");
 
+    // waterfox position control
+    if (TabmixSvc.isWaterfox && Tabmix.isVersion(780)) {
+      MozXULElement.insertFTLIfNeeded("browser/preferences/preferences.ftl");
+      const position = $("tabBarPosition");
+      const menupopup = position.firstChild;
+      while (menupopup.hasChildNodes()) {
+        menupopup.firstChild.remove();
+      }
+      menupopup.appendChild(MozXULElement.parseXULToFragment(
+        `<menuitem value="topAboveAB"
+            data-l10n-id="tab-top-above-ab"/>
+         <menuitem value="topUnderAB"
+            data-l10n-id="tab-top-under-ab"/>
+         <menuitem value="bottom"
+            data-l10n-id="tab-bottom"/>`
+      ));
+      position.setAttribute("preference", "browser.tabBar.position");
+      position.setAttribute("value", Services.prefs.getCharPref("browser.tabBar.position"));
+      const preferences = document.documentElement.querySelector("preferences");
+      preferences.appendChild(MozXULElement.parseXULToFragment(
+        `<preference id="browser.tabBar.position"
+                     name="browser.tabBar.position"
+                     type="string"/>`
+      ));
+    }
+
```

Comment:
> The option dialog window.
> On my system this option have no effect if there is not an opened **about:preference** instance.

@117649 ,
this code work for me, maybe there is some race condition?
I don't know why ```about:preferences``` change anything
check if you can find if calling this code early help

or maybe try to change this
```diff
      position.setAttribute("preference", "browser.tabBar.position");
      position.setAttribute("value", Services.prefs.getCharPref("browser.tabBar.position"));
-     const preferences = document.documentElement.querySelector("preferences");
+     const preferences = $("paneAppearance").querySelector("preferences");
      preferences.appendChild(MozXULElement.parseXULToFragment(
```

Reply:
got it !

```
        `<preference id="browser.tabBar.position"
                     name="browser.tabBar.position"
                     type="string"
                     onchange="window.opener.moveTabBar();"/>`
```